Automattic/woocommerce-payments
# RPP: Handle fraud prevention service in InitialState

Fixes #7413

#### Changes proposed in this Pull Request

1 - Duplicate legacy `Fraud_Prevention_Service` to `src` with the following changes - commit d09463f984861ab01fd546982534956a4d435543, and then integrate it into `InitialState`:  

- Remove `set_instance` and `get_instance` as we no longer need singleton pattern.
- Change constructor args to `SessionService` and `WC_Payments_Account`.
- Simplify `get_token`.
- Correct param type for `verify_token`.

2 - The removal part of the legacy will be handled in Jan 2024 to avoid any potential issue for merchants during high-season in Nov and Dec 2023. PR here: https://github.com/Automattic/woocommerce-payments/pull/7697

3 - Introduce `PaymentProcessingException`, which would be used for other exceptions during the verification state.

#### Testing instructions

- Enable `card_testing_prevention_enabled` on server to activate this feature.
- Add a breakpoint to the new code in `InitialState` to confirm they're really run into.
- Ideally test for both classic and block checkouts.
- To throw exception/error, a simple trick is to run this JS code in browser console: 

```js
document.querySelector('#wcpay-fraud-prevention-token').value += '_change';
``` 





-------------------


**Post merge**
